### PR TITLE
Feauture/signup realization

### PR DESCRIPTION
**Pull request by HIMDanny**

### CODE REVIEW

File: client/src/api/postSignUpData.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  gender,
+  avatarUrl,
+}) => {
+  const body = { firstName, lastName, login, email, password, telephone, birthdate, gender, avatarUrl };
+  try {
+    const { status } = await axios.post('/customers', body);
+
+    return { isSuccess: status === 200 };
+  } catch (err) {
+    return { isSuccess: false, error: err.response.data };
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/api/postSignUpData.js]:
```suggestion
    return { error: err.response.data };
```

**Review by Ahmed-Fahmy-Dev — COMMENTED** [review]
the isSuccess: false is not needed

File: client/src/api/postSignUpData.js (hunk 1 above)
Comment (HIMDanny) [reply client/src/api/postSignUpData.js]: Ack and fixed

**Review by Ahmed-Fahmy-Dev — APPROVED** [review]

File: client/src/features/signup/data/formModel.js [hunk 2]
```diff
@@ -37,13 +37,13 @@ export default {
     invalidErrorMsg: 'Passwords must match',
   },
 
-  phone: {
-    name: 'phone',
+  telephone: {
+    name: 'telephone',
     label: 'Phone',
```
Comment (AltysS) [inline client/src/features/signup/data/formModel.js:42]: Please, make this field as required. I can go ahead and not type any data at this field and then catch an err

image

File: client/src/features/signup/data/formModel.js [hunk 3]
```diff
@@ -37,13 +37,13 @@ export default {
     invalidErrorMsg: 'Passwords must match',
```
Comment (AltysS) [inline client/src/features/signup/data/formModel.js:37]: Please, indicate that password must have more than 7 symbols. 

P.S kindly ask you to hide the password )))) 

image

**Review by AltysS — APPROVED** [review]

**Pull request merged**